src/System.Text.Encoding.CodePages/tests/EncodingCodePages.cs

```diff
@@ -458,20 +458,28 @@ public static void TestDefaultEncodings()
         {
             ValidateDefaultEncodings();
 
-            foreach (object[] mapping in CodePageInfo())
-            {
-                Assert.Throws<NotSupportedException>(() => Encoding.GetEncoding((int)mapping[0]));
-                Assert.Throws<ArgumentException>(() => Encoding.GetEncoding((string)mapping[2]));
-            }
-            // Currently the class EncodingInfo isn't present in corefx, so this checks none of the code pages are present.
-            // When it is, comment out this line and remove the previous foreach/assert.
-            // Assert.Equal(CrossplatformDefaultEncodings, Encoding.GetEncodings().OrderBy(i => i.CodePage).Select(i => Map(i.CodePage, i.WebName)));
-
             // The default encoding should be something from the known list.
             Encoding defaultEncoding = Encoding.GetEncoding(0);
             Assert.NotNull(defaultEncoding);
             KeyValuePair<int, string> mappedEncoding = Map(defaultEncoding.CodePage, defaultEncoding.WebName);
-            Assert.Contains(mappedEncoding, CrossplatformDefaultEncodings());
+
+            if (defaultEncoding.CodePage == Encoding.UTF8.CodePage)
+            {
+                // if the default encoding is not UTF8 that means either we are running on the full framework
+                // or the provider is registered. at that time we shouldn't expect exceptions when creating
```

Comment:
I'll modify the comment to be clearer. thanks

---
In reply to: [95671559](https://github.com/dotnet/corefx/pull/15079#discussion_r95671559) [](ancestors = 95671559)